geth/signal/signals.go

```diff
@@ -39,7 +39,16 @@ type Envelope struct {
 
 // NodeCrashEvent is special kind of error, used to report node crashes
 type NodeCrashEvent struct {
-	Error string `json:"error"`
+	Error error `json:"error"`
+}
+
+// MarshalJSON implements the json.Marshaller interface.
+func (e NodeCrashEvent) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
```

Comment:
Any particular reason to pass by pointer instead of value? It should work the same https://play.golang.org/p/7lBq845BC7k

Reply:
Yep, should switch it back to passing by value. Was originally like this for consistency because I was mistakenly defining MarshalJSON for a pointer. Now it should be by value.  I'll update the PR in a bit.